data/changelog/dune/2024-04-03-dune.3.15.0.md

```diff
@@ -0,0 +1,111 @@
+   
+   - melange: fix inconsistency in virtual library implementation. Concrete
+     modules within a virtual library can now refer to its virtual modules too
+     (#10051, fixes #7104, @anmonteiro)
+   
+   - melange: fix a bug that would cause stale `import` paths to be emitted when
+     moving source files within `(include_subdirs ..)` (#10286, fixes #9190,
+     @anmonteiro)
+   
+   - Dune file formatting: output utf8 if input is correctly encoded (#10113,
+     fixes #9728, @moyodiallo)
+   
+   - Fix expanding dependencies and locks specified in the cram stanza.
+     Previously, they would be installed in the context of the cram test, rather
+     than the cram stanza itself (#10165, @rgrinberg)
+   
+   - Fix bug with `dune exec --watch` where the working directory would always be
+     set to the project root rather than the directory where the command was run
+     (#10262, @gridbugs)
+   
+   - Regression fix: sign executables that are promoted into the source tree
+     (#10263, fixes #9272, @emillon)
+   
+   - Fix crash when decoding dune-package for libraries with `(include_subdirs
+     qualified)` (#10269, fixes #10264, @emillon)
+   
+   ### Changed
+    
+    - Remove the `--react-to-insignificant-changes` option. (#10083, @rgrinberg)
+---
+We're happy to announce that Dune 3.15.0 is now available.
+This feature has many fixes and new features that you can find in the changelog.
+
+There are a few new features that we would like to specially highlight.
+
+### Removal of previous limitations in many forms
+
+Prior to Dune 3.15 there were a number of limitations where percent forms like
+`%{env:...}` could be used to expand to useful values.
+
+In the new version some of these limitations have been lifted, so for example
+`{env:...}` can be used in `install` stanzas.
+
+Likewise there was no consistency where `%{cma:...}` or `%{cmo:...}` could be
+used. With the new code, these forms should work consistently everywhere.
+
+Similarly the variables allowed in `enabled_if` fields have been expanded, from
+just allowing variables that can be computed from the context to now allowing
+all variables as long as expanding these variables does not introduce
+dependency cycles.
+
+These relaxed rules can be combined to enable a library depending on
+environment variables, e.g. `(enabled_if %{env:ENABLE_LIBFOO=false}))`.
+
+### Overlapping names in different contexts
+
+Continuing the theme of conditionally enabling or disabling code to be built,
+Dune 3.15 now allows overlapping names between `executable` and `melange.emit`
+targets. This makes it possible to share the same name in different contexts
+(e.g. one context with native compilation and one emitting code for the
+browser).
+
+### Properly output UTF-8 encoded text when formatting
+
+Dune's own files are assumed to be UTF-8 encoded, yet the formatter would err
+on the safe side and escape bytes outside the ASCII range, even if the files
+were valid UTF-8 sequences.
```

Comment:
I don't think that it's completely accurate. We didn't assume anything about the encoding of dune files (similarly to what the compiler does). What changed is that if the parsed value is valid utf8, we output a utf8-encoded dune file.